dataprep/clean/clean_date_utils.py

```diff
@@ -0,0 +1,683 @@
+        """
+        This function set value of weekday_token
+        Parameters
+        ----------
+        weekday_token
+            token string of weekday
+        """
+        self.weekday_token = weekday_token
+        return True
+
+    def set_timezone_token(self, timezone_token: str) -> bool:
+        """
+        This function set value of timezone_token
+        Parameters
+        ----------
+        timezone_token
+            token string of timezone
+        """
+        self.timezone_token = timezone_token
+        return True
+
+    def set_tzinfo(
+        self,
+        timezone: str = "",
+        utc_add: str = "",
+        utc_offset_hours: int = -1,
+        utc_offset_minutes: int = -1,
+    ) -> bool:
+        """
+        This function set timezone info
+        Parameters
+        ----------
+        timezone
+            name of timezone
+        utc_add
+            the offset is positive or negtive comaring to UTC
+        utc_offset_hours
+            value of offset hours
+        utc_offset_minutes
+            value of offset minutes
+        """
+        if timezone != "":
+            self.tzinfo["timezone"] = timezone
+        if utc_add != "":
+            self.tzinfo["utc_add"] = utc_add
+        if utc_offset_hours >= 0:
+            self.tzinfo["utc_offset_hours"] = utc_offset_hours
+        if utc_offset_minutes >= 0:
+            self.tzinfo["utc_offset_minutes"] = utc_offset_minutes
+        return True
+
+    def set_valid(self, valid: bool) -> bool:
+        """
+        This function set valid status of target format
+        Parameters
+        ----------
+        valid
+            valid status
+        """
+        self.valid = valid
+        return True
+
+    def set_ispm(self, ispm: bool) -> bool:
+        """
+        This function set value of judgement
+        of PM status for target format
+        Parameters
+        ----------
+        ispm
+            If is PM, True. If not, False
+        """
+        self.hms_token["ispm"] = ispm
+        return True
+
+    def add_invalid_token(self, token: str) -> bool:
+        """
+        This function set value of invalid tokens
+        in target format
+        Parameters
+        ----------
+        token
+            invalid token
+        """
+        self.invalid_tokens.append(token)
+        return True
+
+
+def split(txt: Union[str, Any], seps: Union[str, Any]) -> Any:
+    """
+    This function split string into tokens
+    Parameters
+    ----------
+    txt
+        string
+    seps
+        seprators
+    """
+    default_sep = seps[0]
+    for sep in seps[1:]:
+        txt = txt.replace(sep, default_sep)
+    result = [i.strip() for i in txt.split(default_sep)]
+    result = [value for value in result if value != ""]
+    return result
+
+
+def check_date(date: Union[str, Any]) -> Any:
+    """
+    This function check format of date
+    Parameters
+    ----------
+    date
+        date string
+    """
```

Comment:
Can we add more comments for this check function to clarify logic

Reply:
Yeah. I've add more comments ~